### ISSUE TITLE
Two panel view in editor mode is gone for markdown files

### ISSUE THREAD

**Issue by tomposmiko** [go-gitea/gitea#11972]
- Gitea version (or commit ref):
- Git version: 1.12.0
- Operating system: Linux Ubuntu 18.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

On try.gitea.io:
https://try.gitea.io/tompos/aaa/_edit/master/a.md

The two panel editing option is gone. There is no more way to edit a markdown file and see the rendered changes in real time.
...

**Comment by silverwind** [comment]
I wasn't aware of the existance of such a mode but this is intended because markdown files now edit like any other file and it was done in foresight for the SimpleMDE removal. It could of course be introduced again.

**Comment by tomposmiko** [comment]
Yes, please it was an extremely nice editor, actually better the most other editors.

**Comment by tomposmiko** [comment]
I just realized, not only the two panel mode, but the formatting toolbar is also gone.
This is very bad. The user experience is now way worse than before. Why was the change actually? What problem required this solution?

**Comment by tomposmiko** [comment]
Also headings are hard to be distinguished, the font sizes are very close to each other.

**Comment by silverwind** [comment]
We want to remove the bloated, buggy and unmaintained markdown editor, see https://github.com/go-gitea/gitea/issues/10729. Generally I'm not sure if we want to treat editing markdown files differently, GitHub also doesn't.

**Comment by tomposmiko** [comment]
Yes, GitHub also does not provide realtime preview for the .md files in the editor.
So Gitea's editor was way better so far. It's a huge drawback now and it would be a huge improvement and delighter to bring it back.

**Comment by lunny** [comment]
@tomposmiko This changed because we want to edit codes on the web. Maybe we should have a markdown priview plugin on monaco editor.

**Comment by lafriks** [comment]
And adding toolbar back would also be probably doable

**Comment by silverwind** [comment]
I have a hard time understanding how this has worked before (I've never seen it). Was it a side-by-side editor? I guess if the screen is of sufficient width, we could just put the content of the preview tab at the side of the editor instead.

**Comment by mrsdizzie** [comment]
you could still see on the wiki editor since it is still SimpleMDE -- just edit a page and click the 2nd to last button 'toggle side by side'

**Comment by silverwind** [comment]
I see, so it's fullscreening the editor and preview. Certainly possible but I think we should first replace SimpleMDE, add that new editing toolbar and then make a shared abstraction over both textarea-based and monaco-based editor that adds features like this.

**Comment by tomposmiko** [comment]
I don't personally miss the full screen mode, but the real time visualization.
Thanks guys!

I also have concerns about the style (font size?).
Should I report it in a different issue?

**Comment by tomposmiko** [comment]
Well, can we expected any change about this issue?
If not, we can close it.
If yes, that would be awesome!

**Comment by luwol03** [comment]
I will like these feature. Everyone who has the same opinion can react with :thumbsup: emoji.

**Comment by kevintruong** [comment]
I would love to have markdown editor when I edit a markdown file. 
Currently, the default editor is apply for markdown file. 
Is that possible for this one

**Comment by richmahn** [comment]
I was the original developer of the editor and making sure we had a toolbar for properly writing Markdown files, as not everyone knows markdown. I too noticed this missing and found this issue. It would be nice if there was a least a way to turn on or somehow install your own WYSWIG editor.

**Comment by richmahn** [comment]
Why is it the Wiki page for when you can make a new file has a toolbar for Markdown, but not editing a regular .md file in the repo?

**Comment by 73DX** [comment]
Please turn this feature back on. The Markdown tool bar would be a great success to be used in Markdown documents using the Github UI. 
Contributors that way do not need to install additional software of use other sites to do it.

**Comment by davewat** [comment]
We are stuck on version < 1.12.0 until the ability to edit Markdown (with editor toolbar/preview) is re-enabled.

Our use case - we have a large number of users that provide simple edits to MD documentation in a repo.  They are not technical, and cannot handle MD. The toolbar/ preview capability allows them to edit the documentation.

**Comment by silverwind** [comment]
> with editor toolbar/preview

The Markdown preview tab is still there, there's just no side-by-side mode currently.

As for toolbar, I'm thinking about integrating https://github.com/github/markdown-toolbar-element but it may not be easy because the module expects a textarea to work with, so some hackery will probably be required to make it work with monaco.

**Comment by Raytle** [comment]
I would like side-by-side mode.
